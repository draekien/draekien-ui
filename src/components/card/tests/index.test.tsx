import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Card from '..';

describe('Card component', () => {
  test('should match snapshot', () => {
    const { container } = render(<Card>Card</Card>);

    expect(container).toMatchSnapshot();
  });

  test('should have maxWidth 10rem when size is small', () => {
    const { container } = render(
      <Card backgroundColor="b-000" size="small">
        Card
      </Card>
    );

    const styles = window.getComputedStyle(container.firstChild as Element);

    expect(styles.maxWidth).toBe('10rem');
  });

  test('should have maxWidth 40rem when size is large', () => {
    const { container } = render(
      <Card backgroundColor="b-000" size="large">
        Card
      </Card>
    );

    const styles = window.getComputedStyle(container.firstChild as Element);

    expect(styles.maxWidth).toBe('40rem');
  });

  test('should have width 100% when size is large', () => {
    const { container } = render(<Card fullWidth>Card</Card>);

    const styles = window.getComputedStyle(container.firstChild as Element);

    expect(styles.width).toBe('100%');
  });

  test('should call onClick callback function if callback function exists', () => {
    const fn = jest.fn();
    const component = render(<Card onClick={fn}>Card</Card>);

    fireEvent(
      component.getByText('Card'),
      new MouseEvent('click', { bubbles: true, cancelable: true })
    );

    expect(fn).toHaveBeenCalledTimes(1);
  });

  test('should render header and footer', () => {
    const { container } = render(
      <Card cardHeader="header" cardFooter="footer">
        content
      </Card>
    );

    expect(container.textContent).toContain('header');
    expect(container.textContent).toContain('footer');
  });
});
