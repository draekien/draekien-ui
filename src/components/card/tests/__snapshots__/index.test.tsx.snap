// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<div>
  <div
    class="css-15tes2-Card"
  >
    <div
      class="css-8e25xx"
    >
      Card
    </div>
  </div>
</div>
`;

exports[`Card component should render with bottom accent 1`] = `
<div>
  <div
    class="css-15tes2-Card"
  >
    <div
      class="css-8e25xx"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Card component should render with left accent 1`] = `
<div>
  <div
    class="css-1t98pyd-Card"
  >
    <div
      class="css-8e25xx"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Card component should render with right accent 1`] = `
<div>
  <div
    class="css-15tes2-Card"
  >
    <div
      class="css-8e25xx"
    >
      content
    </div>
  </div>
</div>
`;

exports[`Card component should render with top accent 1`] = `
<div>
  <div
    class="css-15az4k1-Card"
  >
    <div
      class="css-8e25xx"
    >
      content
    </div>
  </div>
</div>
`;
